From e38e5682093f0ddfb50ed6c7a17d4842ac7ccd3f Mon Sep 17 00:00:00 2001
From: David Andrey <david.andrey@netmodule.com>
Date: Mon, 22 Jul 2013 16:36:27 +0200
Subject: [PATCH 2/9] ze7000: phy: adapt marvell settings

The onboard PHY is compatible with the 88E1116R,
based on the UID at least
Marvell settings are overwritten to met the board
requirements

Signed-off-by: David Andrey <david.andrey@netmodule.com>

Conflicts:
	arch/arm/boot/dts/ze7000.dts
---
 drivers/net/phy/marvell.c |   19 ++++++++++++++++++-
 1 file changed, 18 insertions(+), 1 deletion(-)

diff --git a/drivers/net/phy/marvell.c b/drivers/net/phy/marvell.c
index 7a7aad1..0d8a6e6 100644
--- a/drivers/net/phy/marvell.c
+++ b/drivers/net/phy/marvell.c
@@ -390,11 +390,28 @@ static int m88e1116r_config_init(struct phy_device *phydev)
	if (err < 0)
		return err;
	temp = phy_read(phydev, MII_M1116R_CONTROL_REG_MAC);
-	temp |= (1 << 5);
+	temp &= ~(1 << 5);  /* @@ NetModule, da, set correct mode for ZE7000 board */
	temp |= (1 << 4);
	err = phy_write(phydev, MII_M1116R_CONTROL_REG_MAC, temp);
	if (err < 0)
		return err;
+
+	/* @@ NetModule, da, LED settings */
+	err = phy_write(phydev, MII_MARVELL_PHY_PAGE, 3);
+	if (err < 0)
+		return err;
+
+	temp = 0x1040;
+	err = phy_write(phydev, 16, temp);
+	if (err < 0)
+		return err;
+
+	temp = 0x4405;
+	err = phy_write(phydev, 17, temp);
+	if (err < 0)
+		return err;
+
+
	err = phy_write(phydev, MII_MARVELL_PHY_PAGE, 0);
	if (err < 0)
		return err;
--
1.7.10.4
